app/src/main/java/com/asfoundation/wallet/verification/intro/VerificationIntroInteractor.kt

```diff
@@ -0,0 +1,77 @@
+package com.asfoundation.wallet.verification.intro
+
+import com.adyen.checkout.core.model.ModelObject
+import com.appcoins.wallet.bdsbilling.WalletService
+import com.appcoins.wallet.billing.adyen.*
+import com.asfoundation.wallet.billing.adyen.AdyenPaymentInteractor
+import com.asfoundation.wallet.support.SupportInteractor
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.Single
+
+class VerificationIntroInteractor(
+    private val adyenPaymentRepository: AdyenPaymentRepository,
+    private val adyenPaymentInteractor: AdyenPaymentInteractor,
+    private val walletService: WalletService,
+    private val supportInteractor: SupportInteractor
+) {
+
+  companion object {
+    val PAYMENT_TYPE = AdyenPaymentRepository.Methods.CREDIT_CARD
+  }
+
+  fun loadVerificationIntroModel(): Single<VerificationIntroModel> {
+    return getVerificationInfo()
+        .flatMap {
+          loadPaymentMethodInfo(it.currency, it.value)
+              .map { paymentInfoModel -> mapToVerificationIntroModel(it, paymentInfoModel) }
+        }
+  }
+
+  fun disablePayments(): Single<Boolean> {
+    return adyenPaymentInteractor.disablePayments()
+  }
+
+  fun makePayment(adyenPaymentMethod: ModelObject, shouldStoreMethod: Boolean,
+                  returnUrl: String): Single<VerificationPaymentModel> {
+    return walletService.getAndSignCurrentWalletAddress()
+        .flatMap {
+          adyenPaymentRepository.makeVerificationPayment(adyenPaymentMethod, shouldStoreMethod,
+              returnUrl, it.address, it.signedAddress)
+        }
+  }
+
+  fun getAuthorisedTransaction(uid: String): Observable<PaymentModel> {
+    return adyenPaymentInteractor.getAuthorisedTransaction(uid)
+  }
```

Comment:
Not used, I think it may really not be needed, but double check it

Reply:
I removed this since it was not being used and didn't seem like it was needed, but I'm not sure what was the original purpose..